src/app/services/mail/mail.service.ts

```diff
@@ -157,13 +164,24 @@ export class MailService {
       })
 
       try {
+        let sendMailFromSG = false
+        const rand = Math.random() * 100
+        sendMailFromSG = rand <= config.nodemailer_client_threshold_sg
```

Comment:
⚠️ This line should use `<` instead of `<=`. The reason for that is `Math.random()` returns values in the range `[0,1[` (greater than or equal to 0 and less than 1 - [doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random))

A strict less-than operator will ensure a threshold of zero will indeed always yield evaluations as `false`, and a threshold of `100` will always yield evaluations as `true`.

I don't remember if we verified that for our react migration threshold checks btw 😰... 

In practice it probably doesn't matter all that much because getting `0` exactly from `Math.random()` practically never happens. But still, for code correctness, the check operation should be `<` instead of `<=`.

Reply:
fixed!